src/string_pool.cr

```diff
@@ -33,8 +33,18 @@ class StringPool
   getter size : Int32
 
   # Creates a new empty string pool.
-  # Default capacity is 256 entries which amounts to 4096 bytes of memory.
-  def initialize(@capacity = 256)
+  #
+  # The *initial_capacity* is useful to avoid unnecessary reallocations
+  # of the internal buffers in case of growth. If you have an estimate
+  # of the maximum number of elements the pool will hold it should
+  # be initialized with that capacity for improved performance.
+  #
+  # ```
+  # pool = StringPool.new(256)
+  # pool.size # => 0
+  # ```
+  def initialize(initial_capacity : Int = 8)
+    @capacity = initial_capacity
```

Comment:
The type restriction is too wide and it won't work if you pass anything other than Int32. Let's make that Int32 instead.